Add bg arctile for capabilities (#221)
sharetribe/flex-docs: src/docs/background/23-how-capabilities-work/index.md

---

> title: How Capabilities work?

Reviewer: This article seems to be discussing the changes and migration, rather than just providing background on the capabilities framework, so I think "Changes to Stripe provider onboarding and Stripe capabilities framework" would be better.

slug: how-capabilities-work
updated: 2019-12-19
category: background
ingress:
  Capabilities framework offers tools to manage regulatory aspects of
  provider onboarding. This article outlines mandatory changes to the
  framework and instructions regarding your actions on reacting to those
  changes.
published: true
---

## Introduction

Coping with regulatory environment regarding provider onboarding can

> be a complex and daunting endeavour. Fulfilling these requirements

Reviewer 2: ```suggestion
be a complex and daunting task. Fulfilling these requirements
```

No need to be any more fancy than what it needed. It just makes reading slower.

means that you need to be collecting a variety of information from
your providers in order to be able to process payments. Stripe has a
mechanism for controlling the amount of information collected from providers called
[Capabilities](https://stripe.com/docs/connect/capabilities-overview).

Capabilities are a collection of settings that can be requested for
each provider. What Capabilities are required determines what
information Stripe requires to be collected from the providers.

## Changes to Capabilities framework

Stripe is introducing mandatory changes to the Capabilities framework.
In the beginning of 2020, Stripe will enforce that Capabilities are

> required for all accounts, existing and new. Stripe’s deadlines for

Reviewer 2: Having both "enforce" and "required" kind of repeats the same thing. How about "Stripe will enforce Capabilities for all accounts" or "Stripe will require Capabilities from all accounts" etc. ?

reacting to these changes are:
- 20th of February, 2020 for customers in the US.

> - 31st of March, 2020 for the rest of the world.

Reviewer 2: I'm actually not sure if that is grammatically correct way to write dates. It is clear, explicit, and understandable for Finns that are used to that format, but maybe double check form a reputable source.

> - In practice, there might be small variations for these depending on the

Reviewer 2: This could be just a paragraph after the bullet point list?

 country and activity of the marketplace.

After requesting new Capabilities, your providers need to most likely
provide additional verification information to Stripe and you need to
update your provider onboarding and payment information updating
processes in your marketplace application.

## How should you handle the changes to Capabilities framework?

For Flex payments to operate smoothly, your accounts need to have the
`card_payments` and `transfers` Capabilities required and activated.
The `card_payments` will enable the providers to accept payments by
card and the `transfers` will allow the platform account to move money
to connected accounts. Requesting `card_payments` will enforce also
requesting the `transfers` capability.

Requesting Capabilities can be done during [Stripe
Account](https://www.sharetribe.com/api-reference/marketplace.html#stripe-account)
[create](https://www.sharetribe.com/api-reference/marketplace.html#create-stripe-account)

> and

Reviewer 2: It is really hard to visually separate the two links next to each other, party because our hover effect is really subtle. Maybe the "Stripe account" link can be removed or just combined to the following "create" link?

[update](https://www.sharetribe.com/api-reference/marketplace.html#update-stripe-account)
calls. Activating the Capabilities happens by providing identity
verification information to Stripe either by using [Stripe Connect

> Onboarding](https://stripe.com/en-fi/connect/onboarding), or [account

Reviewer: Maybe add a paragraph that explains that the recommended approach is to integrate with Connect Onboarding?

tokens](https://stripe.com/docs/connect/account-tokens)
during the create and update calls. What information is required for

> Capability activation is dependent on provider account and activity.

Reviewer: ```suggestion
Capability activation is dependent on the provider's account and activity.
```

> The specific information can be determined from `stripeAccountData` in

Reviewer: Maybe:

The specific list of required information can be programatically determined by examining `stripeAccountData` ...

[Stripe
Account](https://www.sharetribe.com/api-reference/marketplace.html#stripe-account)
resource. The `stripeAccountData` attribute contains the related [Stripe Account

> Object](https://stripe.com/docs/api/accounts/object) where the fields

Reviewer: ```suggestion
Object](https://stripe.com/docs/api/accounts/object), as returned by Stripe's API, where the fields
```

`payouts_enabled` and `requirements` can provide you with valuable information.

> How to implement these changes depends on when you have subscribed to Flex.

Reviewer: How about:

Depending on when you have subscribed to Flex, you may need to take action and implement changes to your marketplace's provider onboarding flow.

> ### Before 19th of December

Author: This date will be updated when we finally deploy changes.

> ### Instructions for Flex subscriptions started before 19th of December 2019

Author: Update the date, when the correct one is know.

If you have subscribed to Flex before 19th of December you need to

> react to the changes. We at Sharetribe have worked hard so that you

Reviewer 2: I hate when people say "worked hard" to others to let the others feel better about themselves, but using the same sentence of yourself feels really off to me. Subjective indeed, and I don't want to diminish the amount of work put here, but somehow this is putting a bit too much focus on who does things and not on what has been done.

can implement the changes as smoothly as possible.

**Migration on 8th of January 2020**

We will perform a migration on 8th of January requiring for all of
your provider accounts `card_payments`, `transfers` and
`legacy_payments` capabilities and updating your marketplace's Stripe
API version to `2019-12-03`. `card_payments` and `transfers`
capabilities are required for payments to work after the above

> deadlines. `legacy_payments` is a capability allowing your current

Reviewer: From customer's point of view, there's single deadline, so I'd use singular here and below.

implementation to work until the deadlines.

**Your actions**

In order for you to make sure that your marketplace is ready for the
above deadlines, you need to perform the following actions:

- If you wish to use Stripe Connect Onboarding, enable it in your
 [Stripe

>  Dashboard](https://dashboard.stripe.com/account/applications/settings)

Reviewer: What if they don't want to use Connect Onboarding? Should we describe briefly what is needed then? Perhaps:

- If you prefer to keep the provider onboarding flow within your marketplace user interface, you may need to implement changes to your provider information forms and make sure that all information required by Stripe is collected.

Author: This is stated in the how-to article.

> - Make sure that your marketplace application is ready for the

Author: Add link to How to when it's done

 changes by following our how to guide for provider onboarding
- Inform your providers that they need to provide additional

>  information to Stripe.

Reviewer 2: This is actually not a simple task to achieve currently. It's doable with the CSV user export or with the Integration API, but should be instruct how that can be done?

You can already start working, implementing and testing the provider
onboarding and updating processes in your marketplace application.
Before deploying these changes to production, we recommend that you
wait for the migration on 8th of January. The migration will make
implementing things simpler regarding your existing provider accounts.

> ### After 19th of December

Author: This date will be updated when we finally deploy changes.

> ### Instructions for Flex subscriptions started after 19th of December 2019

Author: Update the date, when the correct one is know.

Your marketplace by default supports and is ready for the changes to
the Capabilities framework. In practice:

> - Your marketplace is using Stripe API version `2019-12-03`

Reviewer 2: This might easily become out-of-date when we update the backend. Should we not mention the API version here?
